### PR TITLE
feat: add JsonWriterCache.java and added JsonWriterCache in DirectWriter to allow JsonWrites

### PR DESCRIPTION
**Pull request by allenc3**

### CODE REVIEW

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/DirectWriter.java [hunk 1]
```diff
@@ -33,11 +33,12 @@
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.logging.Logger;
+import org.json.JSONArray;
 
 /**
  * Writer that can help user to write data to BigQuery. This is a simplified version of the Write
  * API. For users writing with COMMITTED stream and don't care about row deduplication, it is
- * recommended to use this Writer.
+ * recommended to use this Writer. The DiectWriter can be used to write both JSON and protobuf data.
```
Comment (pmakani) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/DirectWriter.java:41]: DiectWriter -> DirectWriter

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/JsonWriterCacheTest.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+import com.google.protobuf.Timestamp;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.junit.*;
```
Comment (pmakani) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/JsonWriterCacheTest.java:38]: use single class imports

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/DirectWriter.java (hunk 1 above)
Comment (allenc3) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/DirectWriter.java:41]: Got it

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/JsonWriterCacheTest.java (hunk 2 above)
Comment (allenc3) [reply google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/JsonWriterCacheTest.java:38]: Done!

**Pull request merged**